### ISSUE TITLE
Rogue Bugs

### ISSUE THREAD

**Issue by Radekas** [SunwellTracker/issues#329]
Decription: 
Rogue's have plenty of bugs, but the ones that appear to be most obvious and abusable are: The ability to loot _while_ in stealth, and Envenoming an Immune target refreshes SnD while not consuming combo points (Assass).

How it works: 
1. Rogue kills mobs, forgets to loot > Dies. Resses near enemies > Stealth > Loots corpses free of charge while in stealth. (Since Angrather Launch)

2. *E.G.* Second boss in AK (Ahn'Khet): *Add is up, boss is immune* Uses envenom while SnD is dipping low > SnD refreshes (combo points not consumed) > DPS and downtime on this bossfight is completely removed in comparison to other ramping classes.

How it should work: 

1. It shouldn't.
2. A. Either combo points are consumed and deals 0 damage and does not refresh SnD, or B. SnD is refreshed but the combo points are consumed. (Preferably option A)

Source (you should point out proofs of your report, please give us some source):

**Comment by Radekas** [comment]
OH. And a MAJOR one if not already fixed. If the Rogue is in group combat which it did not initiate (E.G. Tank Pulls 5 mobs in a dungeon) The rogue has the ability to Stealth After it kills each target it has chosen to enter combat with (providing they have not used aoe abilitys or been attacked by more than 1 mob)
TL;DR : *5 mobs* Stealth > Kill > Stealth > Kill > Stealth > Kill, etc etc.

**Comment by sitowsky** [comment]
There is no source in this report. Please find a source(proof) and report bug properly once again. 

Thanks in advance.
